src/local_package.ml

```diff
@@ -142,3 +182,15 @@ let virtual_lib t = Lazy.force t.virtual_lib
 
 let meta_template t =
   Path.extend_basename (meta_file t) ~suffix:".template"
+
+let defined_in sctx ~dir =
+  let local_packages = of_sctx sctx in
+  let by_build_dir =
```

Comment:
Should `by_build_dir` become a part of a memoized value?
I'm not sure how often this is called. Is it once for each directory?
If we have a big repo with many hundreds of directories and packages, that might add up?

Reply:
Yeah, I think that's good. Should we just memoize generating the `Path.Map.t` itself however?

We could also make accessing it a memoized function as well. That will give us another improvement for cases where the map changes but the key we are interested in does not.